q2_quality_control/tests/test_decontam.py

```diff
@@ -115,6 +115,63 @@ def test_combined(self):
                 expecter_table = biom.Table.from_tsv(th, None, None, None)
             self.assertEqual(test_table, expecter_table)
 
+    def test_freq_prev_control_col_error_raised(self):
+        with self.assertRaises(ValueError) as context:
```

Comment:
There's a method called `assertRaisesRegex` that takes two arguments, the error that is expected to be raised, and a regex string that is expected to match the error message. For example, in this case you could use:

```python
with self.assertRaisesRegex(ValueError, '--prev-control-column'):
    ...
```

and get the same functionality as the `assertRaises` and `assertIn` assertions combined. (I think your approach is very clear as well, so no need to refactor, but just thought I'd let you know.)

Reply:
Thanks for the info!